### ISSUE TITLE
`ghq list` stops list when non-directory encountered

### ISSUE THREAD

**Issue by jps3** [motemen/ghq#63]
The `ghq list [<query>]` command stops listing matches when a non-directory. This is not a show-stopper, but it was making me think I had lost my mind this morning, when I would successfully `ghq get …` then a `ghq list jps3/` would stop listing items after the first few. Sometime in the past few days it appears I mistakenly edited a text file in the `~/.ghq/github.com/jps3/` directory, and I noticed that in normal sorting it came right after the first few items which were being returned with `ghq list jps3/`. Once I removed that file, all was right in my `ghq` world again.

Found/tested on OS X 10.10.5 (14F27) and `ghq` version 0.7.1 (installed via Homebrew).

Example:
```
$ ghq list motemen/
github.com/motemen/ghq
$ cd ~/.ghq/github.com/motemen
$ ls -l
total 0
drwxr-xr-x  27 jps3  staff  918 Sep 23 10:12 ghq
$ touch foo
$ ls -l 
total 0
-rw-r--r--   1 jps3  staff    0 Sep 23 10:14 foo
drwxr-xr-x  27 jps3  staff  918 Sep 23 10:12 ghq
$ ghq list motemen/
$ 
$ rm foo
$ ghq list motemen/
github.com/motemen/ghq
$ touch zoo
$ ls -l
total 0
drwxr-xr-x  27 jps3  staff  918 Sep 23 10:12 ghq
-rw-r--r--   1 jps3  staff    0 Sep 23 10:16 zoo
$ ghq list motemen/
github.com/motemen/ghq
$ rm zoo
$ mkdir foo.d
$ ghq list motemen/
github.com/motemen/ghq
$ rmdir foo.d
```

Perhaps someone else has run into this and been a bit flummoxed as well. Also, maybe this is an easy one to fix (if so, you are welcome).

**Comment by kyanny** [comment]
Similar to https://github.com/motemen/ghq/pull/61.
Seems happened after https://github.com/motemen/ghq/pull/54 (IMHO this behaviour is confusing)

**Comment by lestrrat** [comment]
I had a file like $ghq_root/github.com/hoge.go, and all repositories that sort alphabetically AFTER that filename did not show. MASSIVELY CONFUSING.

The culprit is this clause here. This is just silly. Just misplacing a file *completely* fails ghq.
https://github.com/motemen/ghq/blob/master/local_repository.go#L148-L152

**Comment by lestrrat** [comment]
@motemen I really really want #54 to be reverted. this behavior is really confusing.
if I have `$ghq_root/github.com/hoge.go` placed, every directory that comes after hoge.go (such as `github.com/lestrrat/*` fails to show up on `ghq list`. This the fact the existence of a single unrelated file changes the behavior of ghq is unacceptable to me.

**Comment by motemen** [comment]
Thank you and sorry for the delayed response, I will fix this issue in a couple of days.

**Comment by motemen** [comment]
Just released [v0.7.2](https://github.com/motemen/ghq/releases/tag/v0.7.2) including #66 , which should fix the issue. Please try the newest version, and sorry for the inconvenience.

**Comment by kyanny** [comment]
:tada:
